### PR TITLE
[Typescript All] Prevent usage of reserved keyword for model fields

### PR DESCRIPTION
**Pull request by JFCote**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [x] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09)

### Description of the PR

Reserved keywords were escaped everywhere but for this. I just add it in the `AbstractTypescriptClientCodegen`. I run all the typescript sample I could find. All the changes are not related to my change because the petstore doesn't contain any reserved keyword.

### CODE REVIEW

**Comment by TiFu** [comment]
Just to check that I understand the issues this PR solves correctly: does this change escape Enum values, which are reserved keywords? e.g. escapes `await` in an enum with value `await`

**Comment by JFCote** [comment]
I think enum were already escaped (I did not validate). I'm escaping model members which are reserved keywords.

For example, if you have a model called `Dog` with a boolean named `default`, it will escape `default` with `_default` in the final generated code.

Without this escape, the typescript is not compiling.

Does it make more sense? :)

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java [hunk 1]
```diff
@@ -405,6 +405,10 @@ public String toEnumName(CodegenProperty property) {
                 if (Boolean.TRUE.equals(var.isEnum)) {
                     var.datatypeWithEnum = var.datatypeWithEnum.replace(var.enumName, cm.classname + "." + var.enumName);
                 }
+
+                if (isReservedWord(var.name)) {
```
Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java]: @JFCote Shall we do it in `toVarName` instead similar to other generators? e.g. https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/RubyClientCodegen.java#L453

Comment (JFCote) [reply modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java]: @wing328 As you can see here: https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java#L169

In typescript, the toVarName is simply returning the variable name itself without transformation. Should I change it to the same thing as javascript? 
https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavascriptClientCodegen.java#L490

Let me know and I'll do the changes!

Comment (wing328) [reply modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java]: Please follow the JS approach, which is used by other generators (e.g. Ruby, PHP, etc)

Comment (JFCote) [reply modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java]: Ok! Will try to commit a fix this week! Thanks for the review!

**Comment by JFCote** [comment]
@wing328 Changes are done! Let me know if something is still missing!

**Comment by wing328** [comment]
@JFCote saw your change. Do we still need the reserved word check in `postProcessModels`?

**Comment by JFCote** [comment]
@wing328 Do you think it's called from somewhere else automatically?

**Comment by JFCote** [comment]
@wing328 I've just test it and you are right, it seems to be called from somewhere, so no need in the `postProcessModels`. I've just submit a new version.

**Comment by wing328** [comment]
@JFCote looks good to me 👍

**Comment by kenisteward** [comment]
Thanks for this.

**Comment by gennadiylitvinyuk** [comment]
The config Parameter modelPropertyNaming is now ignored

**Pull request merged**